bitsandbytes/cextension.py

```diff
@@ -298,7 +298,15 @@ def get_native_library() -> BNBNativeLibrary:
 
 
 try:
-    lib = get_native_library()
+    if hasattr(torch, "xpu") and torch.xpu.is_available():
+        if not ipex_xpu:
+            logger.warning(
+                "Detected Intel XPU without intel_extension_for_pytorch installed. Triton implementation will be used."
+            )
+        # create a mock with error messaging as fallback
+        lib = ErrorHandlerMockBNBNativeLibrary("XPU does not need library loading")
```

Comment:
Wouldn't we still want it in case the user has a reason to use the two CPU ops it implements? We don't necessarily know they won't just because an XPU is available.